src/libs/language-javascript-latest/_babelHelpers.js

```diff
@@ -0,0 +1,84 @@
+
+//
+// Transpiling with babel is needed because describeApi will directly
+// call `require(PROJECT_ROOT + '/api');
+//
+
+import path from 'path';
+import { stripIndent } from 'common-tags';
+import { debug, warning } from '../../logger';
+
+// Support for `babel` property in project package.json to extend base babel config
+// to add plugins and more presets, etc. React, JSX Transform, Syntax support
+const babelRequiredPkgJson = require(path.join(process.cwd(), 'package.json'));
+
+// check if babel key is package.json
+const hasBabelConfigInPkgJson = ('babel' in babelRequiredPkgJson);
+
+if (hasBabelConfigInPkgJson) {
+  warning(stripIndent`
+    You have specified a custom babel configuration; please make sure
+    that you are compiling code for the correct version of Node.js (6.10).
+    Usually you may want to include 'babel-preset-dawson'.
+  `);
+}
+
+// if key, use that babel config, add babelrc = false
+// else, use default
+export const BABEL_CONFIG = hasBabelConfigInPkgJson
+  ? Object.assign({}, babelRequiredPkgJson['babel'], {
```

Comment:
`{}` as the first parameter to Object.assign to prevent overwriting properties of `babelRequiredPkgJson`